mcpserver/config.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2023-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package mcpserver
+
+import (
+	"context"
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+// Context key types
+type contextKey string
+
+const (
+	UserIDKey contextKey = "userID"
+	TokenKey  contextKey = "token"
+)
+
+// Config represents the configuration for the MCP server
+type Config struct {
+	// Mattermost server URL (e.g., "https://mattermost.company.com")
+	ServerURL string `json:"server_url"`
+
+	// Personal Access Token for authentication
+	PersonalAccessToken string `json:"personal_access_token"`
+
+	// Optional headers to include in requests
+	Headers map[string]string `json:"headers,omitempty"`
+
+	// Timeout for requests to Mattermost
+	RequestTimeout time.Duration `json:"request_timeout"`
+
+	// Transport type (currently only stdio is supported)
+	Transport string `json:"transport"`
+
+	// Development mode enables additional tools for setting up test data
+	DevMode bool `json:"dev_mode"`
+}
+
+// AuthenticationProvider handles authentication for MCP requests
+type AuthenticationProvider interface {
```

Comment:
This doesn't feel like it belongs in config.
It also feels like it's not generic it's for tokens specifically. Presumably this is where you mean to hook in OAuth Later?